webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -268,36 +266,58 @@ class TechniqueWriter (
+      res <- r.await
+      _   <- ZIO.when(res.code != 0) {
+        Inconsistency(
+          s"An error occurred when translating technique.json file into Rudder language\n code: ${res.code}\n stderr: ${res.stderr}\n stdout: ${res.stdout}"
+        ).fail
+      }
+      r <- RunNuCommand.run(Cmd("/opt/rudder/bin/rudderc", "compile" :: "-j" :: "-f" :: "\"cf\"" :: "-i" :: s"\"${basePath}/${techniquePath(technique)}/technique.rl\"" :: s"--config-file=${configFile}" :: Nil, Map.empty))
+      res <- r.await
+      _   <- ZIO.when(res.code != 0) {
+        Inconsistency(
+          s"An error occurred when compiling technique.rl file into cfengine\n code: ${res.code}\n stderr: ${res.stderr}\n stdout: ${res.stdout}"
+        ).fail
       }
-      case None => ApplicationLogger.error(s"PATH environment variable must be defined to run rudder-lang test loop.")
+      r <- RunNuCommand.run(Cmd("/opt/rudder/bin/rudderc", "compile" :: "-j" :: "-f"  :: "\"dsc\"" :: "-i" :: s"\"${basePath}/${techniquePath(technique)}/technique.rl\"" :: s"--config-file=${configFile}" :: Nil, Map.empty))
+      res <- r.await
+      _   <- ZIO.when(res.code != 0) {
+        Inconsistency(
+          s"An error occurred when compiling technique.rl file into dsc\n code: ${res.code}\n stderr: ${res.stderr}\n stdout: ${res.stdout}"
+        ).fail
+      }
+    } yield  {
+      ()
     }
   }
 
   // Write and commit all techniques files
   def writeTechnique(technique : Technique, methods: Map[BundleName, GenericMethod], modId : ModificationId, committer : EventActor) : IOResult[Technique] = {
     for {
-      agentFiles <- writeAgentFiles(technique, methods, modId, committer)
       metadata   <- writeMetadata(technique, methods, modId, committer)
       // Before writing down technique, set all resources to Untouched state, and remove Delete resources, was the cause of #17750
-      updateResources  = technique.ressources.collect{case r if r.state != ResourceFileState.Deleted => r.copy(state = ResourceFileState.Untouched) }
-      techniqueWithResourceUpdated = technique.copy(ressources = updateResources)
+      updateResources = technique.ressources.collect{case r if r.state != ResourceFileState.Deleted => r.copy(state = ResourceFileState.Untouched) }
+                         techniqueWithResourceUpdated = technique.copy(ressources = updateResources)
       json       <- writeJson(techniqueWithResourceUpdated)
+      agentFiles <- callRudderc(technique).catchAll(e =>
+        for {
+          _ <- ApplicationLoggerPure.info(s"An error occurred when compiling technique '${technique.name}' (id : '${technique.bundleName}') with rudderc, error details will be displayed below, falling back to old saving process")
+          _ <- ApplicationLoggerPure.error(s"Rudderc error when compiling technique '${technique.name}' (id : '${technique.bundleName}'): ${e.fullMsg}")
+          - <-  writeAgentFiles (technique, methods, modId, committer)
```

Comment:
Nice :)
Perhaps we should store these rudderc exception in dedicated, independendant log files, for ex in `/var/log/rudder/rudderc/failures/YYYY-MM-dd-hhmmss-techniqueName.log` with the context that lead to the error:
- json file
- if it exists, `cf` `dsc` file
- the error log

That would allow users to send us these reports (or even to get them automatically with the support script)